examples/Bickley_jet_shallow_water.jl

```diff
@@ -0,0 +1,244 @@
+        u = ComputedField(uh / h)
+        v = ComputedField(vh / h)
+  ω_field = ComputedField( ∂x(vh/h) - ∂y(uh/h) )
+   ω_pert = ComputedField( ω_field - Ω )
+
+# ## Running a `Simulation`
+#
+# We pick the time-step that ensures to resolve the surface gravity waves.
+# A time-step wizard can be applied to use an adaptive time step.
+
+simulation = Simulation(model, Δt = 1e-2, stop_time = 150.00)
+
+# ## Prepare output files
+#
+# Define a function to compute the norm of the perturbation on the cross channel velocity.
+# We obtain the `norm` function from `LinearAlgebra`.
+
+using LinearAlgebra, Printf
+
+function perturbation_norm(model)
+    compute!(v)
+    return norm(interior(v))
+end
+nothing # hide
+
+# Choose the two fields to be output to be the total and perturbation vorticity.
+
+outputs = (ω_total = ω_field, ω_pert = ω_pert)
+
+# Build the `output_writer` for the two-dimensional fields to be output.
+# Output every `t = 1.0`.
+
+simulation.output_writers[:fields] =
+    NetCDFOutputWriter(
+        model,
+        (ω = ω_field, ωp = ω_pert),
+        filepath=joinpath(@__DIR__, "Bickley_jet_shallow_water.nc"),
+        schedule=TimeInterval(1.0),
+        mode = "c")
+
+# Build the `output_writer` for the growth rate, which is a scalar field.
+# Output every time step.
+#
+
+simulation.output_writers[:growth] =
+    NetCDFOutputWriter(
+        model,
+        (perturbation_norm = perturbation_norm,),
+        filepath=joinpath(@__DIR__, "perturbation_norm_shallow_water.nc"),
+        schedule=IterationInterval(1),
+        dimensions=(perturbation_norm=(),),
+        mode = "c")
+
+run!(simulation)
+
+# ## Visualize the results
+#
+
+using NCDatasets
+nothing # hide
+
+# This sets the environment to avoid an error in GKSwstype
+
+ENV["GKSwstype"] = "100"
+
+# Define the coordinates for plotting
+
+xf = xnodes(ω_field)
+yf = ynodes(ω_field)
+nothing # hide
+
+# Define keyword arguments for plotting the contours
+
+kwargs = (
+         xlabel = "x",
+         ylabel = "y",
+           fill = true,
+         levels = 20,
+      linewidth = 0,
+          color = :balance,
+       colorbar = true,
+           ylim = (0, Ly),
+           xlim = (0, Lx)
+)
+nothing # hide
+
+# Read in the `output_writer` for the two-dimensional fields
+# and then create an animation showing both the total and perturbation
+# vorticities.
+
+using Plots
+
+ds = NCDataset(simulation.output_writers[:fields].filepath, "r")
+
+iterations = keys(ds["time"])
+
+anim = @animate for (iter, t) in enumerate(ds["time"])
+     ω = ds["ω"][:, :, 1, iter]
+    ωp = ds["ωp"][:, :, 1, iter]
+
+     ω_max = maximum(abs, ω)
+    ωp_max = maximum(abs, ωp)
+
+     plot_ω = contour(xf, yf, ω',  clim=(-ω_max,  ω_max),  title=@sprintf("Total ω at t = %.3f", t); kwargs...)
+    plot_ωp = contour(xf, yf, ωp', clim=(-ωp_max, ωp_max), title=@sprintf("Perturbation ω at t = %.3f", t); kwargs...)
+
+    plot(plot_ω, plot_ωp, layout = (1,2), size=(1200, 500))
```

Comment:
```suggestion
      ω = ds["ω"][:, :, 1, iter]
     ωp = ds["ωp"][:, :, 1, iter]

     ωp_max = maximum(abs, ωp)

     plot_ω = contour(xf, yf, ω',
                      clim = (-1, 1),
                     title = @sprintf("Total ω at t = %.3f", t); kwargs...)
    
    plot_ωp = contour(xf, yf, ωp',
                      clim = (-ωp_max, ωp_max),
                     title = @sprintf("Perturbation ω at t = %.3f", t); kwargs...)

    plot(plot_ω, plot_ωp, layout = (1, 2), size=(1200, 500))
```

Reply:
I fixed the `clim` for the total vorticity to +/-1. I think it looks nicer! What do you think @francispoulin?

Reply:
Much better I agree!

Reply:
Much better

Reply:
But @francispoulin, I also suggested much more changes (reformatting of the plotting to avoid clutter...).
You disagree with those or missed it?